src/Arcus.Security.Tests.Unit/Core/CachedSecretProviderTests.cs

```diff
@@ -247,5 +247,35 @@ public async Task CachedSecretProvider_GetTwoSecretsAsync_SkippedCache_ShouldRet
             Assert.Equal(expectedSecondSecret, actualSecond.Value);
             Assert.Equal(2, spyTestProvider.CallsMadeSinceCreation);
         }
+
+        [Fact]
+        public async Task CacheSecretProvider_GetSecretAsync_InvalidateSecret_ShouldRequestNewSecretEvenWithinCacheDuration()
+        {
+            // Arrange
+            string expectedFirstSecret = Guid.NewGuid().ToString("N");
+            string expectedSecondSecret = Guid.NewGuid().ToString("N");
+            var testProviderStub = new TestSecretProviderStub(expectedFirstSecret);
+            var cacheInterval = TimeSpan.FromSeconds(10);
+
+            const string keyName = "MySecret";
+            var cachedSecretProvider = new CachedSecretProvider(
+                testProviderStub,
+                new CacheConfiguration(cacheInterval));
+
+            Secret actualFirst = await cachedSecretProvider.GetSecretAsync(keyName);
+            testProviderStub.SecretValue = expectedSecondSecret;
+
+            // Act
+            await cachedSecretProvider.InvalidateSecretAsync(keyName);
```

Comment:
Should we do a new get secret to see if it first caches it correct, or do you rely on the other tests for that?

Reply:
Yeah, I thought of other scenarios; but I guess all those are already covered.

Reply:
Are they already covered or should we still add it? Just want to make sure it's covered 😅

Reply:
I think that we have all the useful scenario's covered.